.github/workflows/appengine-cleanup.yml

```diff
@@ -0,0 +1,29 @@
+name: Automated AppEngine cleanup script
+on:
+  # Run once every week on Sunday
+  schedule:
+    - cron: "0 0 * * 0"
+  # To manually trigger workflow
+  workflow_dispatch: {}
+jobs:
+  appengine_cleanup:
+    runs-on: ubuntu-latest
+    env:
+      CI_OPTIONS: '--quiet'
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    steps:
+      - uses: actions/checkout@v3
+      - name: 'Authenticate with Google Cloud'
+        uses: 'google-github-actions/auth@v1'
+        with:
+          workload_identity_provider: 'projects/1038484894585/locations/global/workloadIdentityPools/github-wi-pool/providers/github-wi-provider'
+          service_account: 'appengine-cleanup-non-prod@bvdp-saturn-dev.iam.gserviceaccount.com'
+      - name: 'Set up Cloud SDK'
+        uses: 'google-github-actions/setup-gcloud@v1'
+      # Delete old AppEngine versions in each environment
+      - run: |
```

Comment:
What do you think about making each environment a separate step with [continue-on-error](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error) set to true? I think that would make it easier to see if the script did fail in some environment.

Reply:
I investigated this, but I wasn't sure that if it continued on error that it would report the overall status of the pipeline to success instead of failure.

Reply:
Does it really matter what the workflow's overall status is? It's not gating anything else.

As is, it looks like the workflow would always report success, even if the script failed in one or more environments. I think it would be nice, but not necessary, if the workflow was marked as failed if the script failed.

Reply:
You could make them separate jobs, and then set them to always run. If any fails the overall workflow will still fail.

For example: https://github.com/broadinstitute/rawls/blob/develop/.github/workflows/rawls-run-azure-e2e-tests.yaml#L94

Reply:
> Does it really matter what the workflow's overall status is? It's not gating anything else.
> 
> As is, it looks like the workflow would always report success, even if the script failed in one or more environments. I think it would be nice, but not necessary, if the workflow was marked as failed if the script failed.

It's not gating anything else, but it will clutter up people's emails and notifications with spam for failed jobs due to no items to clean up, which doesn't seem ideal.

Reply:
I disagree, I don't think it's an issue with the script but an issue with our expectations.

In manual use, the user expectation for a delete operation should be that if any abnormal situation is encountered, then the operation is aborted and the user is informed of the issue. I would classify trying to do a delete operation and having nothing to delete as an abnormal situation and one that you would definitely want the user to notice. It could be that despite the abnormal situation you don't want to take any action, but it also could be that your assumptions were mistaken and you need to investigate why there was nothing to delete in the first place.

The difference is that now the script is automated and there is no user intervention. Errors are routine in a server environment (consider 400s). You probably want them logged, but don't want to act on them unless something is truly broken.

I don't think there are any errors that we would want to act on in the script in an automated environment:

    The service account has the right permissions, so that error won't trigger
    gcloud is installed by the previous step
    jq is installed by default on ubuntu-latest
    all the other aborts are related to the number of deployments that are left

I am okay with switching to continue-on-error and seeing how much noise that generates. Then worst case we switch our current process from "I ran this script manually" to "I checked the error in the GitHub Action run and there was nothing to do".